plugins/hls-module-name-plugin/src/Ide/Plugin/ModuleName.hs

```diff
@@ -118,22 +129,28 @@ action state uri =
 -- | Possible module names, as derived by the position of the module in the
 -- source directories.  There may be more than one possible name, if the source
 -- directories are nested inside each other.
-pathModuleNames :: IdeState -> NormalizedFilePath -> String -> IO [T.Text]
-pathModuleNames state normFilePath filePath
+pathModuleNames :: Recorder (WithPriority Log) -> IdeState -> NormalizedFilePath -> FilePath -> IO [T.Text]
+pathModuleNames recorder state normFilePath filePath
   | isLower . head $ takeFileName filePath = return ["Main"]
   | otherwise = do
       session <- runAction "ModuleName.ghcSession" state $ use_ GhcSession normFilePath
       srcPaths <- evalGhcEnv (hscEnvWithImportPaths session) $ importPaths <$> getSessionDynFlags
-      paths <- mapM makeAbsolute srcPaths
+      logWith recorder Debug (ModuleNameLog $ "ModuleName.srcpath: " <> T.pack (unlines srcPaths))
+
+      paths <- mapM canonicalizePath srcPaths
+      logWith recorder Debug (ModuleNameLog $ "ModuleName.paths: " <> T.pack (unlines paths))
```

Comment:
Canonicalisation is, in my experience, too big of a hammer and behaves unexpectedly with symbolic links.

We basically removed its usage except for in tests and non-lsp usage. See https://github.com/haskell/haskell-language-server/pull/2359 and https://github.com/haskell/haskell-language-server/issues/2358 where we ran into issues. 

Would you mind looking into this test-case and check this patch doesn't cause any regressions?

Reply:
I checked the project in #2358, it works fine for the current patch.

But to reduce concerns further, I'll try to investigate `normalise`, it seems workable.